imap/annotate.c

```diff
@@ -1200,22 +1202,30 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
     frock.since_modseq = since_modseq;
     frock.flags = flags;
 
-    r = _annotate_getdb(mailbox ? mailbox_uniqueid(mailbox) : NULL, mailbox, uid, 0, &frock.d);
+    /* special case where mboxname is "" and others are empty, this means
+     * server annotations, which are recognised by an empty pattern by the
+     * code below */
+    if (!mailbox && !pattern && !*mboxname) {
+        mailbox = NULL;
```

Comment:
Same tests still failing... ah, looks like a typo.  This fixes it:

```suggestion
        mboxname = NULL;
```